app/controllers/users_controller.rb

```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = 'User created.'
+      log_in @user
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @organized_events = @user.organized_events
+    redirect_to root_url and return unless @user == current_user
```

Comment:
* [ ] list events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] list upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

Reply:
thanks done with all the requirement .